### PR TITLE
test: add creating/spending validity checks for rare output scripts

### PR DESCRIPTION
**Pull request by theStack**
This PR adds a functional test with a collection of output scripts that are "rare"/obscure, in the sense that they wouldn't be used or make sense within a regular payment transaction, but creating them is still considered standard, which is often considered surprising and counter-intuitive. They either
* use a weird pubkey encoding (-> hybrid pubkeys [in types P2PK, P2MS]),
* are provably unspendable (-> not-on-curve pubkeys [in types P2PK, P2MS, P2TR]), or
* are spendable by anyone with current consensus rules (-> future segwit versions)

For all of those, the creation is considered standard, whereas spending for some is only consensus-valid; for not-on-curve pubkeys, spending is obviously impossible for mathematical reasons. The behaviour for future segwit versions is already tested in `feature_segwit.py`, so there is a certain overlap, but I think it stlll makes thematically sense to collect them here as well (if this PR gets Concept ACKed in general). OP_RETURN outputs don't fall into this "rare" category as they are more or less widespread and it's well-known that they can be used to embed arbitrary data (we already test them in `mempool_datacarrier.py`). Happy to include other examples as well if I missed any.

### CODE REVIEW

File: test/functional/feature_rare_output_scripts.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            self.wallet.send_to(from_node=self.node, scriptPubKey=p2ms_script, amount=21000)
+
+        self.log.info("Check that creating P2TR outputs with not-on-curve pubkey is standard")
+        self.wallet.send_to(from_node=self.node, scriptPubKey=output_key_to_p2tr_script(invalid_xonly_pub), amount=21000)
+
+    def test_future_segwit_versions(self):
+        self.log.info("Check that creating future segwit outputs (2 <= version <= 16) is standard")
+        funding_txs = []
+        for future_ver in range(2, 16+1):
+            witness_program = random.randbytes(random.randrange(2, 40+1))
```
Comment (tdb3) [inline test/functional/feature_rare_output_scripts.py:124]: Thinking out loud:
Initially thought of ensuring the edge cases (2 and 40) are explicitly added in addition to random program size selection, but this would probably be overkill.

File: test/functional/feature_rare_output_scripts.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        tx.vin = [CTxIn(COutPoint(int(outpoint[0], 16), outpoint[1]))]
+        tx.vout = [CTxOut(40000, self.wallet.get_scriptPubKey())]
+        tx.wit.vtxinwit.append(CTxInWitness())
+        return tx
+
+    def test_hybrid_pubkeys(self):
+        self.log.info("Check that creating hybrid pubkey P2PK outputs is standard")
+        privkey_even, pubkey_even = self.generate_uncompressed_keypair(is_even=True)
+        assert pubkey_even.p.y.is_even()
+        pubkey_even_script = key_to_p2pk_script(b'\x06' + pubkey_even.get_bytes()[1:])
```
Comment (tdb3) [inline test/functional/feature_rare_output_scripts.py:69]: Thinking out loud:
Might be nice to enhance ECPubKey to handle hybrid keys (prevents having to do byte manipulation here).  Seems outside the scope of this PR though.

**Review by tdb3 — APPROVED** [review]
ACK 006d835acf98ebceaaaead7aea41f512ed5023ad
Seems like a good addition.  Observed a runtime of around 1s, so minimal.
Left some thoughts.  Will also think about other potential candidate rare output scripts and comment if any come to mind.

File: test/functional/feature_rare_output_scripts.py (hunk 2 above)
Comment (instagibbs) [reply test/functional/feature_rare_output_scripts.py:69]: not sure it's worth the effort for such a niche thing that already has unit test coverage

**Review by instagibbs — COMMENTED** [review]
~0 on this file tbh

it's kind of a hodgepodge of things being tested that might already have functional test coverage, like future witness spends which is an intended upgrade hook and should be well-tested?

If they already don't have functional coverage maybe there are better spots for each?
